files/scripts/syncd.sh

```diff
@@ -13,25 +9,25 @@ function debug()
 
 function lock_service_state_change()
 {
-    debug "Locking ${LOCKFILE} from ${SERVICE} service"
+    debug "Locking ${LOCKFILE} from ${SERVICE}$DEV service"
 
     exec {LOCKFD}>${LOCKFILE}
     /usr/bin/flock -x ${LOCKFD}
     trap "/usr/bin/flock -u ${LOCKFD}" 0 2 3 15
 
-    debug "Locked ${LOCKFILE} (${LOCKFD}) from ${SERVICE} service"
+    debug "Locked ${LOCKFILE} (${LOCKFD}) from ${SERVICE}$DEV service"
 }
 
 function unlock_service_state_change()
 {
-    debug "Unlocking ${LOCKFILE} (${LOCKFD}) from ${SERVICE} service"
+    debug "Unlocking ${LOCKFILE} (${LOCKFD}) from ${SERVICE}$DEV service"
     /usr/bin/flock -u ${LOCKFD}
 }
 
 function check_warm_boot()
 {
-    SYSTEM_WARM_START=`sonic-db-cli STATE_DB hget "WARM_RESTART_ENABLE_TABLE|system" enable`
-    SERVICE_WARM_START=`sonic-db-cli STATE_DB hget "WARM_RESTART_ENABLE_TABLE|${SERVICE}" enable`
+    SYSTEM_WARM_START=`/usr/bin/sonic-db-cli $DEV STATE_DB hget "WARM_RESTART_ENABLE_TABLE|system" enable`
+    SERVICE_WARM_START=`/usr/bin/sonic-db-cli $DEV STATE_DB hget "WARM_RESTART_ENABLE_TABLE|${SERVICE}" enable`
```

Comment:
>$DEV [](start = 46, length = 4)

Could you use `ip netns exec` directly?

Reply:
This is done so that it is generic for single or multi-asic. $DEV will be passed only for multi-asic. For single asic platform, as there is no namespace created, ip netns exec cannot be used.

Reply:
I see your concern. There are several workarounds:
1. [https://serverfault.com/a/627634](https://serverfault.com/a/627634) will create a `default` namespace
2. Since the context is bash, you can create a `function` to handle the logic instead of only one variable `DEV`.

---
In reply to: [383602697](https://github.com/Azure/sonic-buildimage/pull/3888#discussion_r383602697) [](ancestors = 383602697)